# Supermarket Sales Dataset:  Exploratory Data Analysis Exercise

In [1]:
import pandas as pd

In [3]:
# import dataset in JN
sales = pd.read_csv('supermarket_sales.csv')

#### Display the first 5 rows of the dataframe

In [4]:
sales.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total  \
0       Health and beauty       74.69       7.0  26.1415  548.9715   
1  Electronic accessories       15.28       5.0   3.8200   80.2200   
2      Home and lifestyle       46.33       7.0  16.2155  340.5255   
3       Health and beauty       58.22       8.0  23.2880  489.0480   
4       Sports and travel       86.31       7.0  30.2085  634.3785   

         Date   Time      Payment  
0  01/05/2019  13:08      Ewallet  
1  03/08/2019  10:29         Cash  
2  03/03/2019  13:23  Credit card  
3  01/27/2019  20:33      Ewallet  
4  02/08/2019  10:37      Ewallet

#### Check the basic details

In [5]:
sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1007 entries, 0 to 1006
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Invoice ID     1000 non-null   object 
 1   Branch         1000 non-null   object 
 2   City           1000 non-null   object 
 3   Customer type  1000 non-null   object 
 4   Gender         1000 non-null   object 
 5   Product line   1000 non-null   object 
 6   Unit price     1000 non-null   float64
 7   Quantity       1000 non-null   float64
 8   Tax 5%         1000 non-null   float64
 9   Total          1000 non-null   float64
 10  Date           1000 non-null   object 
 11  Time           1000 non-null   object 
 12  Payment        1000 non-null   object 
dtypes: float64(4), object(9)
memory usage: 102.4+ KB


Based from the dataset information, the Invoice ID shoud be in string  data type while Date and Time are should be in date time data type 

In order to change this data types we need to convert these columns in their respective data types

In [13]:
# convert into string
sales['Invoice ID'] = sales['Invoice ID'].astype(str)

#  Converting object to datetime
sales['Date'] = pd.to_datetime(sales['Date'], format = '%m/%d/%Y')

#  Converting object to datetime
sales['Time'] = pd.to_datetime(sales['Time'], format = '%H:%M')

sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1007 entries, 0 to 1006
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Invoice ID     1007 non-null   object        
 1   Branch         1000 non-null   object        
 2   City           1000 non-null   object        
 3   Customer type  1000 non-null   object        
 4   Gender         1000 non-null   object        
 5   Product line   1000 non-null   object        
 6   Unit price     1000 non-null   float64       
 7   Quantity       1000 non-null   float64       
 8   Tax 5%         1000 non-null   float64       
 9   Total          1000 non-null   float64       
 10  Date           1000 non-null   datetime64[ns]
 11  Time           1000 non-null   datetime64[ns]
 12  Payment        1000 non-null   object        
dtypes: datetime64[ns](2), float64(4), object(7)
memory usage: 102.4+ KB


The data types in our dataset is now corrected. 

#### We can proceed to the number of rows and columns

In [14]:
sales.shape

(1007, 13)

We have 1007 rows and 13 rows in our dataset

### Data Cleaning

#### Investigate for null values

In [16]:
sales.isnull().sum()

Invoice ID       0
Branch           7
City             7
Customer type    7
Gender           7
Product line     7
Unit price       7
Quantity         7
Tax 5%           7
Total            7
Date             7
Time             7
Payment          7
dtype: int64

In [18]:
sales[sales['Branch'].isna()]

Invoice ID Branch City Customer type Gender Product line  Unit price  \
65         nan    NaN  NaN           NaN    NaN          NaN         NaN   
72         nan    NaN  NaN           NaN    NaN          NaN         NaN   
80         nan    NaN  NaN           NaN    NaN          NaN         NaN   
93         nan    NaN  NaN           NaN    NaN          NaN         NaN   
100        nan    NaN  NaN           NaN    NaN          NaN         NaN   
114        nan    NaN  NaN           NaN    NaN          NaN         NaN   
125        nan    NaN  NaN           NaN    NaN          NaN         NaN   

     Quantity  Tax 5%  Total Date Time Payment  
65        NaN     NaN    NaN  NaT  NaT     NaN  
72        NaN     NaN    NaN  NaT  NaT     NaN  
80        NaN     NaN    NaN  NaT  NaT     NaN  
93        NaN     NaN    NaN  NaT  NaT     NaN  
100       NaN     NaN    NaN  NaT  NaT     NaN  
114       NaN     NaN    NaN  NaT  NaT     NaN  
125       NaN     NaN    NaN  NaT  NaT     NaN

In [22]:
sales = sales.dropna() # remove the null values
sales.isnull().sum()

Invoice ID       0
Branch           0
City             0
Customer type    0
Gender           0
Product line     0
Unit price       0
Quantity         0
Tax 5%           0
Total            0
Date             0
Time             0
Payment          0
dtype: int64

We sucessfully removed the null values on our dataset.

#### Display the updated basic details of the dataframe

In [23]:
sales.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1000 entries, 0 to 1006
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Invoice ID     1000 non-null   object        
 1   Branch         1000 non-null   object        
 2   City           1000 non-null   object        
 3   Customer type  1000 non-null   object        
 4   Gender         1000 non-null   object        
 5   Product line   1000 non-null   object        
 6   Unit price     1000 non-null   float64       
 7   Quantity       1000 non-null   float64       
 8   Tax 5%         1000 non-null   float64       
 9   Total          1000 non-null   float64       
 10  Date           1000 non-null   datetime64[ns]
 11  Time           1000 non-null   datetime64[ns]
 12  Payment        1000 non-null   object        
dtypes: datetime64[ns](2), float64(4), object(7)
memory usage: 109.4+ KB
